### DISCUSSION TITLE
revealjs: format of date entry using r system on title page does not work

### DISCUSSION THREAD

**Discussion by elcansaito** [quarto-dev/quarto-cli#7164] [Q&A] [answered]
### Description

````qmd

---
title: "Test date"
author: JIS
date: "Last compiled on r format(Sys.time(), '%d %B, %Y')"
title-slide-attributes:
data-background-image: "https://raw.githubusercontent.com/elcansaito/images/main/Background.png"
#data-background-opacity: "0.5"
format:
revealjs:
self-contained: true
slide-number: true
width: 1600
height: 900
css: ["theme.css"]
theme: simple
editor: source
execute:
echo: true
editor_options:
chunk_output_type: inline

---
````

Using the following does not work as well. 

date: "`r Sys.Date()`"
date-format: "YYYY-MMM-D"

It only works if you use

date: 2023/10/11
date-format: "YYYY/MMMM/DD"

But then I cannot add the superscript (th) on the 11.

Any help would be appreciate it!! THANKS

**Comment by mcanouil** [comment]
See <https://quarto.org/docs/reference/dates.html>.
You need to read the page down to the bottom where you'll learn what you need to (formatting and escaping).

Using inline R expression is an rmarkdown thing that you should not use in Quarto.

**Comment by elcansaito** [comment]
Thanks for your answer.  However, it is a little strange because this code it works perfectly using the R system.

````qmd
---
title: "Test2"
author: JIS
date: "`r Sys.Date()`"
date-format: "YYYY-MMM-D"
title-slide-attributes: 
  data-background-image: "https://raw.githubusercontent.com/TheRocinante-lab/ImagesPresentation/main/CBGPBackground.png"
  #data-background-opacity: "0.5"
format: 
  revealjs:
    chalkboard: true
    slide-number: true
    width: 1600
    height: 900
    css: ["theme.css"]
    theme: simple
  pdf: default
editor: source
execute: 
  echo: true
editor_options: 
  chunk_output_type: inline

---

````

Maybe you can let me know why?
Thanks

**Reply by mcanouil** [reply, marked as answer]
Because it relies on knitr, but be aware that this kind of things are very fragile and most of the time it is not something that was planned ed to work ... it simply does, thus for dates, since Quarto has a pretty robust date system which includes formatting, there is absolutely no reason to rely on this rmarkdown trick.

**Comment by elcansaito** [comment]
I hear you!!. Thanks for the feedback, appreciate it. Now it works with
````qmd
date: 2023/10/11
date-format: "YYYY, MMM Do"
````

Although I can´t use the subscript, but close enough. Thanks again.

**Reply by mcanouil** [reply]
You can use JavaScript for that.

````qmd
---
title: "Quarto Playground"
format: html
date: today
date-format: 'dddd[, the] Do [of] MMMM, YYYY'
include-after-body:
  - text: |
      <script type="text/javascript">
      const dateElements = document.querySelectorAll("p.date, div.listing-date");
      dateElements.forEach((el) => {
        el.innerHTML = el.innerHTML.replace(
          /(\d+)(rd|st|th)/g,
          "$1<sup style='font-size:0.5em;font-style:italic;'>$2</sup>"
        );
      });
      </script>
---

This is a playground for Quarto.
````

**Reply by elcansaito** [reply]
Really brilliant!!! Thanks a million, mcanouil!!!!
